File: sample/flutter_base/packges/page_router/lib/src/PageRouter.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:service_interface/service_interface.dart';

class Pagerouter implements PageRouterProtocol {
  void onLoad() { }

  void onUnLoad() { }

  void openUrl(Uri uri) {
    print('welcome to page router');
  }

  void pushNamed(BuildContext context, String routeName, {Object? arguments}) {
    Navigator.pushNamed(context, routeName, arguments: arguments);
  }
}

File: output/src/services/PageRouter.ts
import { PageRouterProtocol } from './PageRouterProtocol';
import { createNavigationContainerRef } from '@react-navigation/native';

const navigationRef = createNavigationContainerRef<any>();

export class PageRouter implements PageRouterProtocol {
  onLoad(): void {}

  onUnLoad(): void {}

  openUrl(url: string): void {
    console.log('welcome to page router');
  }

  pushNamed(routeName: string, params?: any): void {
    if (navigationRef.isReady()) {
      navigationRef.navigate(routeName, params);
    }
  }
}